packages/go/index.ts

```diff
@@ -534,8 +531,8 @@ export async function build({
 async function renameHandlerFunction(fsPath: string, from: string, to: string) {
   let fileContents = await readFile(fsPath, 'utf8');
 
-  const fromRegex = new RegExp(`\\b${from}\\b`, 'g');
-  fileContents = fileContents.replace(fromRegex, to);
+  const fromRegex = new RegExp(String.raw`func (${from})\(`, 'g');
```

Comment:
The simpler version before would rename parts it shouldn't which is obvious in retrospect. The updated tests confirms that will not happen going forward.

The right balance for this replacement is important. Is this current form too restrictive? @cb1kenobi @Ethan-Arrowood @styfle @TooTallNate

Reply:
From what I know about Go, I can't say if it's too restrictive from a language standpoint, but I believe your change is certainly safer than a naive word boundary based replacement.

The only thing I'd say is the parens around the `${from}` are unnecessary and can lead to confusion, so they could be omitted:

```javascript
const fromRegex = new RegExp(String.raw`func ${from}\(`, 'g');
```

Reply:
I think the correct thing to do here is switch from regex to parsing the AST. We should already have it from `analyze.go`, right?

Reply:
That would be ideal, but is that reasonable? Wouldn't `analyze.go` need to be reworked to rewrite the function name as it walks the AST and then serialize the tree back to disk? I'm sure it's cheap to do, but I'd guess the level of effort is more than a simple string replacement in Node.

Reply:
Right, I dont know the effort to write the AST back to a string

Reply:
I agree that the best solution is to use the parsed AST, change it, and write out the result. I just don't know nearly enough Go to handle that. We asked around about doing this before with resident Go devs, but they were pretty busy with other work. I think we can ask again now.

I just updated this to be more lax. The new form is ` ${from}\b`, which accounts for all known syntax variations.